### PR TITLE
[MPQEENABLE-306] Refactor ocp_base to make dynamic_client and k8s_client singleton.

### PR DESCRIPTION
**Pull request by tapatel**
Note : Since other classes inherit ocp_base, So make changes accordingly in all relevant classes.

### CODE REVIEW

File: piqe_ocp_lib/tests/resources/test_ocp_base.py [hunk 1]
```diff
@@ -1,36 +1,108 @@
...
+
+        for thread in threads:
+            thread.join()
+
+        dynamic_client1, dynamic_client2 = thread_queue.queue
+        logger.info("Dynamic_Client1 : %s", dynamic_client1)
+        logger.info("Dynamic_Client2 : %s", dynamic_client2)
+        assert dynamic_client1 is dynamic_client2
+
+    # def test_dynamic_client_singleton_for_ocp3x(self, get_kubeconfig_3x):
```
Comment (ascerra) [inline piqe_ocp_lib/tests/resources/test_ocp_base.py:56]: what is the reason that all these methods are all commented out?

Comment (tapatel) [reply piqe_ocp_lib/tests/resources/test_ocp_base.py:56]: Good question...All these commented methods requires more than 1 kubeconfig file as you can see. But currently our upstream CI jenkins only support/provides 1 kubeconfig file. For that reason i have commented out these new methods. 
However, i have tested locally with two kubeconfig file and it works fine.

Comment (tapatel) [reply piqe_ocp_lib/tests/resources/test_ocp_base.py:56]: I believe upstream repo is only for ocp4x plus. We/Team decided to use downstream repo for ocp3x(legacy) usecases. So you are correct, This won't work for ocp3x. We have to use downstream repo and i am not pushing this changes to downstream repo. Please correct me.

Comment (ascerra) [reply piqe_ocp_lib/tests/resources/test_ocp_base.py:56]: That all makes sense, thank you for the explanation!

**Comment by shwetha-h-p** [comment]
+1

**Pull request merged**